src/GitHub.Exports/Services/GitCloneCommandLineService.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Interop;
+using GitHub.Primitives;
+using Rothko;
+
+namespace GitHub.Services
+{
+    [Export(typeof(IGitCloneCommandLineService))]
+    public class GitCloneCommandLineService : IGitCloneCommandLineService
+    {
+        IVsAppCommandLine vsAppCommandLine;
+        IVSGitServices vsGitServices;
+        IVSServices vsServices;
+        IOperatingSystem operatingSystem;
+
+        [ImportingConstructor]
+        internal GitCloneCommandLineService(IGitHubServiceProvider sp, IVSGitServices vsGitServices,
+            ITeamExplorerServices teamExplorerServices, IVSServices vsServices, IOperatingSystem operatingSystem)
+            : this(sp.TryGetService<IVsAppCommandLine>(), vsGitServices, teamExplorerServices, vsServices, operatingSystem)
+        {
+        }
+
+        public GitCloneCommandLineService(IVsAppCommandLine vsAppCommandLine, IVSGitServices vsGitServices,
+            ITeamExplorerServices teamExplorerServices, IVSServices vsServices, IOperatingSystem operatingSystem)
+        {
+            this.vsAppCommandLine = vsAppCommandLine;
+            this.vsGitServices = vsGitServices;
+            this.vsServices = vsServices;
+            this.operatingSystem = operatingSystem;
+        }
+
+        public string FindGitCloneOption()
+        {
+
+            if (vsAppCommandLine == null)
+            {
+                return null;
+            }
+
+            int isPresent;
+            string optionValue;
+            if (ErrorHandler.Failed(vsAppCommandLine.GetOption("GitClone", out isPresent, out optionValue)))
+            {
+                return null;
+            }
+
+            if (isPresent == 0)
+            {
+                return null;
+            }
+
+            return optionValue;
+        }
+
+        public bool TryOpenRepository(string cloneUrl)
```

Comment:
Wouldn't this method be better in `IVSServices`? We already have a `TryOpenRepository` there, it would make sense to have an overload that takes a `UriString` (which is what this is really taking in), and run this logic there.